### PR TITLE
do not print the FSTDEP009 warning in test but use assertion

### PR DESCRIPTION
**Pull request by mcollina**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by zekth — APPROVED** [review]
lgtm

File: test/internals/initialConfig.test.js [hunk 1]
```diff
@@ -10,6 +10,8 @@ const split = require('split2')
 const deepClone = require('rfdc')({ circles: true, proto: false })
 const { deepFreezeObject } = require('../../lib/initialConfigValidation').utils
 
+process.removeAllListeners('warning')
```
Comment (airhorns) [inline test/internals/initialConfig.test.js:13]: Is this gonna cause an annoying bug where other things that attach warning listeners get them removed unexpectedly by this file?

Comment (jsumners) [reply test/internals/initialConfig.test.js:13]: Each test file is an independent process.

Comment (airhorns) [reply test/internals/initialConfig.test.js:13]: Ah! That mitigates cross test listener fuckery, would fastify itself ever add listeners though?

**Review by RafaelGSS — APPROVED** [review]
LGTM.

File: test/internals/initialConfig.test.js (hunk 1 above)
Comment (jsumners) [reply test/internals/initialConfig.test.js:13]: Only through the deprecation library.

**Pull request merged**